### PR TITLE
Desktop environment work

### PR DESCRIPTION
**Pull request by leocoan**
# Please make sure all below checkboxes in the Mandatory section have been checked before submitting your PR (also don't forget to pay attention to the Depending on Circumstances section)

## Mandatory

- [x] The commit message is an imperative and starts with a capital letter (for example, `Add something`, `Remove something`, `Fix something`, etc.).
- [x] I made sure I hadn't put a period at the end of the the commit message(s).

## Depending on Circumstances

Some of the items in the section become mandatory if you are a first-time contributor and/or your changes are relevant to the items.

- [x] Adding a new parameter I didn't forget to reflect it in `README.md`.
- [x] Adding a new parameter or modifying an existing one I didn't break the alphabetical order.
- [ ] Adding a new utility written in Python I didn't forget to reflect it in `pieman/README.md`.
- [ ] Adding a new utility written in Python I didn't forget to increase the version number of the [pieman](https://pypi.org/project/pieman/) package in `pieman/setup.py`, `pieman/pieman/__init__.py` and `essentials.sh`.
- [x] (for first-time contributors) One of the commits in the pull request adds me to the **Acknowledgements** section in `AUTHORS.md`.

### CODE REVIEW

File: bootstrap/12-pm.sh [hunk 1]
```diff
@@ -119,6 +119,14 @@ if [ ! -z "${includes}" ]; then
     done
 fi
 
+if ${DESKOP_ENV}; then
```
Comment (denismosolov) [inline bootstrap/12-pm.sh]: This should be `XFCE4`

File: helpers/base.sh [hunk 2]
```diff
@@ -158,7 +158,7 @@ check_ownership_format() {
 check_pieman_version() {
     local pieman_version=(0 0)
     local output=""
-
+
```
Comment (denismosolov) [inline helpers/base.sh]: Removing this change from here will make change history a little bit clear

File: helpers/pm.sh [hunk 3]
```diff
@@ -80,6 +80,14 @@ install_packages() {
             # shellcheck disable=SC2086
             DEBIAN_FRONTEND=noninteractive chroot_exec apt-get -y -q -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" ${PM_OPTIONS} install "$@"
         else
+            if  ${XFCE4}; then 
+                echo "keyboard-configuration  keyboard-configuration/layout  select  Russian" | chroot_exec debconf-set-selections
+                # add_package_to_includes xfce4
```
Comment (denismosolov) [inline helpers/pm.sh]: please remove commented lines

File: helpers/pm.sh [hunk 4]
```diff
@@ -80,6 +80,14 @@ install_packages() {
             # shellcheck disable=SC2086
             DEBIAN_FRONTEND=noninteractive chroot_exec apt-get -y -q -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" ${PM_OPTIONS} install "$@"
         else
+            if  ${XFCE4}; then
```
Comment (denismosolov) [inline helpers/pm.sh]: Lets move this block outside of install_packages

File: README.md [hunk 5]
```diff
@@ -291,6 +291,10 @@ Specifies the projects location. By default, the directory named `build` is crea
 
 Restricts Pieman to only creating a chroot environment based on the operating system specified via `OS`. The chroot environment is stored in `build/${PROJECT_NAME}/chroot` and can be used immediately or later to reduce the time of building images. See `BASE_DIR`.
 
+##### XFCE4="false"
+
+If this parameter is set to true, then the image will be compiled together with xfce4 and lxdm. If the parameter is set to false, the desktop environment will not be set. At the moment, the ability to create an image with XFCE is only on Ubuntu 18.04.
```
Comment (denismosolov) [inline README.md]: I'd use Ubuntu 18.04 «Bionic Beaver»

File: helpers/pm.sh [hunk 6]
```diff
@@ -86,6 +86,19 @@ install_packages() {
...
+# Configures the keyboard-configuration package.
+# Globals:
+#     None
+# Arguments:
+#     None
+# Returns:
+#     None
+configure_keyboard_configuration() {
+    if is_debian_based; then
+        echo "keyboard-configuration  keyboard-configuration/layout  select  English" | chroot_exec debconf-set-selections
```
Comment (eugulixes) [inline helpers/pm.sh]: Please remove redundant spaces between the words.

File: pieman.sh [hunk 7]
```diff
@@ -207,6 +209,11 @@ check_required_files
 
 split_os_name_into_pieces
 
+if [ "${OS}" == "alpine-3.9-armhf" ] || [ "${OS}" == "debian-buster-armhf" ] || [ "${OS}" == "devuan-jessie-armhf" ] || [ "${OS}" == "kali-rolling-armhf" ] || [ "${OS}" == "raspbian-stretch-armhf" ] || [ "${OS}" == "ubuntu-xenial-armhf" ]; then
```
Comment (eugulixes) [inline pieman.sh]:
```
if ${XFCE4}; then
    desktop_is_supported=false
    for supported_os in ubuntu-bionic-armhf ubuntu-bionit-arm64; do
        if [[ "${OS}" == "${supported_os}" ]]; then
            desktop_is_supported=true
            break
        fi
    done

    if not ${desktop_is_supported}; then
        fatal "Desktop environment is not supported for the chosen OS."
    fi
fi
```

File: pieman.sh [hunk 8]
```diff
@@ -50,6 +50,8 @@ def_bool_var CREATE_ONLY_CHROOT false
 
 def_var DEVICE "rpi-3-b"
 
+def_bool_var XFCE4 false
```
Comment (eugulixes) [inline pieman.sh]: Please put the param in its place according to the alphabetical order.

File: bootstrap/40-deskop.sh [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+if  ${XFCE4}; then
```
Comment (eugulixes) [inline bootstrap/40-deskop.sh]: Remove a redundant space after `if`.

**Pull request merged**